### PR TITLE
fix(smart-actions): transform legacy widgets in hooks

### PR DESCRIPTION
**Pull request by rap2hpoutre**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: src/utils/widgets.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  'text input': 'text editor',
+};
+
+/**
+ * Convert V1 widgets from smart actions to V2 widgets.
+ * V1 widgets still need to be supported as their usage in smart action is legacy.
+ *
+ * @param {*} field A smart action field
+ */
+function setFieldWidget(field) {
```
Comment (rap2hpoutre) [inline src/utils/widgets.js:54]: Please note that `setFieldWidget` mutates the original field. This is not optimal, still, it has been kept like this to be a mirror of what has been done in `forestadmin-server`.

File: src/services/smart-action-hook.js [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+   * Transform fields from an array to an object to ease usage in hook,
+   * adds null value, prepare widgets.
+   *
+   * @param {*} fields A smart action field
+   */
+  getFieldsForUser(fields) {
+    return fields.reduce((previous, current) => {
+      // Update widget from legacy to current format.
+      this.setFieldWidget(current);
+      // Return the field(with a default value to null when none is provided.
```
Comment (GuillaumeCisco) [inline src/services/smart-action-hook.js:17]:
```suggestion
      // Return the field(with a default value set to null when none is provided).
```

**Review by GuillaumeCisco — DISMISSED** [review]
Great!!!
LGTM! 💖

**Review by GuillaumeCisco — APPROVED** [review]
🐞

**Pull request merged**